internal/pkg/v2/ui/http/handle/usecase.go

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ * Copyright 2020 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package handle
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/concurrent"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/application"
+	dtoBase "github.com/edgexfoundry/edgex-go/internal/pkg/v2/application/dto/v2dot0/base"
+	dtoError "github.com/edgexfoundry/edgex-go/internal/pkg/v2/application/dto/v2dot0/error"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/ui/http/routing"
+)
+
+// UseCaseRequest decodes, processes, and returns a result for a use-case-specific endpoint's request.
+func UseCaseRequest(w http.ResponseWriter, r *http.Request, version, kind, action string, router *routing.Router) {
+	defer func() {
+		if r.Body != nil {
+			_ = r.Body.Close()
+		}
+	}()
+
+	routable, exists := router.FindRoute(version, kind, action)
+	if !exists {
+		http.Error(w, "route not found", http.StatusNotFound)
+		return
+	}
+
+	request := routable.Request()
+	if request == nil {
```

Comment:
I think this is answered by https://github.com/edgexfoundry/edgex-go/pull/2285#discussion_r373496056 ...?

Reply:
routable.Request() returns a new empty Request correct? So why would it ever be nil? And why would you want to execute on a nil Request.

Reply:
Yes, routable.Request() returns a new empty Request (that's a use-case specific concrete structure or nil).  

A nil request DTO simply specifies that the use-case doesn't expect or require a request DTO.  In this edge case, you'd still want to execute the invoked use-case but wouldn't want to attempt to unmarshal into a DTO because a DTO isn't expected or wanted.

Resolve if this addresses your question; ask another question if it does not.

Reply:
Ok, that makes sense. All the example so far returned empty Request DTOs, so was expectaing that to be the standard, i.e. never a nil.

Please consider adding a comment with this `why` explanation